src/platform/user/authentication/components/DowntimeBanner.js

```diff
@@ -5,36 +5,80 @@ import environment from 'platform/utilities/environment';
 import { getStatusFromStatuses } from '../constants';
 
 export default function DowntimeBanners() {
-  const { loading, statuses = [] } = useSelector(
+  const { loading, statuses = [], maintenanceWindows = [] } = useSelector(
     state => state.externalServiceStatuses,
   );
   const dispatch = useDispatch();
   const isLocalhost = useMemo(() => environment.isLocalhost(), []);
+
   useEffect(() => {
     if (!loading && !isLocalhost) {
       dispatch(getBackendStatusAction());
     }
-  }, []); // only on load
+  }, []);
+
```

Comment:
A couple of things from the below code.
- Try to mitigate the usage of `let` variables, try to use `const` as much as possible
- Never try to mess with JavaScript dates, use a library like `date-fns` which is tree-shakeable and easier to work with
- Move helper functions like the `checkMaintenanceWindow` and `formattedMaintenanceStatus` to a helpers file then just pass the parameters you need to keep the component readable.